mod-audit-server/src/main/java/org/folio/rest/impl/AuditDataAcquisitionImpl.java

```diff
@@ -63,26 +64,61 @@ public void getAuditDataAcquisitionOrderLineById(String orderLineId, String sort
-        orderLineAuditEventsService.getAuditEventsByOrderLineId(orderLineId, sortBy, sortOrder.name(), limit, offset, tenantId)
-          .map(GetAuditDataAcquisitionOrderLineByIdResponse::respond200WithApplicationJson)
-          .map(Response.class::cast)
-          .otherwise(this::mapExceptionToResponse)
-          .onComplete(asyncResultHandler);
-      } catch (Exception e) {
-        LOGGER.warn("Failed to get order line audit events by order line id: {}", orderLineId, e);
-        asyncResultHandler.handle(Future.succeededFuture(mapExceptionToResponse(e)));
-      }
-    });
+    try {
+      orderLineAuditEventsService.getAuditEventsByOrderLineId(orderLineId, sortBy, sortOrder.name(), limit, offset, tenantId)
+        .map(GetAuditDataAcquisitionOrderLineByIdResponse::respond200WithApplicationJson)
+        .map(Response.class::cast)
+        .otherwise(this::mapExceptionToResponse)
+        .onComplete(asyncResultHandler);
+    } catch (Exception e) {
+      LOGGER.error("Failed to get order line audit events by order line id: {}", orderLineId, e);
+      asyncResultHandler.handle(Future.succeededFuture(mapExceptionToResponse(e)));
+    }
+
+  }
+
+  @Override
+  public void getAuditDataAcquisitionPieceById(String pieceId, String sortBy, AuditDataAcquisitionPieceIdGetSortOrder sortOrder,
+                                               int limit, int offset, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    LOGGER.debug("getAuditDataAcquisitionOrderById:: Retrieving Audit Data Acquisition Piece By Id : {}", pieceId);
+    String tenantId = TenantTool.tenantId(okapiHeaders);
+
+    try {
+      pieceAuditEventsService.getAuditEventsByPieceId(pieceId, sortBy, sortOrder.name(), limit, offset, tenantId)
+        .map(GetAuditDataAcquisitionPieceByIdResponse::respond200WithApplicationJson)
+        .map(Response.class::cast)
+        .otherwise(this::mapExceptionToResponse)
+        .onComplete(asyncResultHandler);
+    } catch (Exception e) {
+      LOGGER.error("Failed to get piece audit events by piece id: {}", pieceId, e);
+      asyncResultHandler.handle(Future.succeededFuture(mapExceptionToResponse(e)));
+    }
+  }
+
+  @Override
+  public void getAuditDataAcquisitionPieceStatusChangeHistoryById(String pieceId, String sortBy,
+                                                                  AuditDataAcquisitionPieceIdStatusChangeHistoryGetSortOrder sortOrder,
+                                                                  int limit, int offset, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    LOGGER.debug("getAuditDataAcquisitionOrderById:: Retrieving Audit Data Acquisition Piece with unique status By Id : {}", pieceId);
+    String tenantId = TenantTool.tenantId(okapiHeaders);
+
+    try {
+      LOGGER.warn("Trying to get piece audit events with unique status by piece id: {}", pieceId);
```

Comment:
It should not be warn

Reply:
can I remove, this message, because it is already exists in debug starting of method

Reply:
sure